applications/nrf5340_audio/src/modules/led.c

```diff
@@ -31,19 +28,12 @@ LOG_MODULE_REGISTER(led, CONFIG_LOG_LED_LEVEL);
 #define CORE_APP_STR "CORE_APP"
 #define CORE_NET_STR "CORE_NET"
 
-#define LABEL_AND_COMMA(node_id) DT_LABEL(node_id),
-#define GPIO_AND_COMMA(node_id) DT_GPIO_PIN(node_id, gpios),
-#define GPIO_LABEL_AND_COMMMA(node_id) DT_GPIO_LABEL(node_id, gpios),
-#define GPIO_FLAGS_AND_COMMMA(node_id) DT_GPIO_FLAGS(node_id, gpios),
+#define GPIO_DT_SPEC_GET_AND_COMMA(node_id) GPIO_DT_SPEC_GET(node_id),
 
 /* The following arrays are populated compile time from the .dts*/
-const char *led_labels[] = { DT_FOREACH_CHILD(DT_PATH(leds), LABEL_AND_COMMA) };
-
-const int led_pins[] = { DT_FOREACH_CHILD(DT_PATH(leds), GPIO_AND_COMMA) };
-
-const char *led_gpio_labels[] = { DT_FOREACH_CHILD(DT_PATH(leds), GPIO_LABEL_AND_COMMMA) };
-
-const int led_gpio_flags[] = { DT_FOREACH_CHILD(DT_PATH(leds), GPIO_FLAGS_AND_COMMMA) };
+static const struct device leds[] = {
+	DT_FOREACH_CHILD(DT_PATH(leds), GPIO_DT_SPEC_GET_AND_COMMA)
+};
```

Comment:
FYI: raised this proposal upstream: https://github.com/zephyrproject-rtos/zephyr/pull/47342